goupnp.go

```diff
@@ -125,15 +125,25 @@ func DeviceByURL(ctx context.Context, loc *url.URL) (*RootDevice, error) {
 // but should not be changed after requesting clients.
 var CharsetReaderDefault func(charset string, input io.Reader) (io.Reader, error)
 
-// HTTPClient is the client used inside of requestXml for fetching XML from a specific endpoint.
-var HTTPClient = http.DefaultClient
+// contextKey is an unexported type which prevents construction of other contextKeys.
+type httpContextKey struct{}
+
+// WithHTTPClient returns a context wrapping ctx with the HTTP client set to c.
+// c must be non-nil.
+func WithHTTPClient(ctx context.Context, c *http.Client) context.Context {
+	return context.WithValue(ctx, httpContextKey{}, c)
+}
 
 func requestXml(ctx context.Context, url string, defaultSpace string, into interface{}) error {
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
 	if err != nil {
 		return err
 	}
-	resp, err := HTTPClient.Do(req)
+	httpClient := ctx.Value(httpContextKey{}).(*http.Client)
```

Comment:
this crashes.

see: https://play.golang.org/p/_d5GtS9MV-s

you want comma-ok form on the LHS

Reply:
`httpClient, _ := ctx.Value(httpContextKey{}).(*http.Client)`